### PR TITLE
Escape passwords in URLs for dump_config

### PR DESCRIPTION
**Pull request by Khrol**
## What changes were proposed in this pull request?

- Hide passwords in connection strings in `dump_config`

## How was this patch tested?

- Unit test is added (I hope CI runs it)
- Tested on local env
image

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Comment by Khrol** [comment]
I don't see any feedback from contributors here. 🐼

**Comment by bjornalm** [comment]
@Khrol Thanks for contributing. I'm sorry that you haven't received any feedback on this yet, we are working on improving the communication with our community. 

@ranade1 Can you have a look at this?

File: desktop/core/src/desktop/api2_tests.py [hunk 1]
```diff
@@ -189,6 +189,19 @@ def test_get_hue_config_private(self):
     assert_true(len(response.content) < len(private_response.content))
 
 
+  def test_url_password_hiding(self):
```
Comment (Harshg999) [inline desktop/core/src/desktop/api2_tests.py:192]: Nice!

**Review by Harshg999 — APPROVED** [review]
LGTM! Thanks for the fix @Khrol!

**Review by bjornalm — APPROVED** [review]
Looks good to me! @Khrol Thanks again for contributing, it is highly valued :-)

**Pull request merged**